silkrpc/commands/engine_api_test.cpp

```diff
@@ -132,4 +135,107 @@ TEST_CASE("handle_engine_get_payload_v1 fails with invalid amount of params", "[
     context_pool_thread.join();
 }
 
+TEST_CASE("handle_engine_new_payload_v1 succeeds if request is expected payload status", "[silkrpc][engine_api]") {
+    SILKRPC_LOG_VERBOSITY(LogLevel::None);
+
+    BackEndMock backend;
+    EXPECT_CALL(backend, engine_new_payload_v1(testing::_)).WillOnce(InvokeWithoutArgs(
+        []() -> asio::awaitable<PayloadStatus> {
+            co_return PayloadStatus{
+                .status = "INVALID",
+                .latest_valid_hash = 0x0000000000000000000000000000000000000000000000000000000000000040_bytes32,
+                .validation_error = "some error"
+            };
+        }
+    ));
+
+    std::unique_ptr<ethbackend::BackEnd> backend_ptr(&backend);
+
+    nlohmann::json reply;
+    nlohmann::json request = R"({
+        "jsonrpc":"2.0",
+        "id":1,
+        "method":"engine_newPayloadV1",
+        "params":[{
+            "parentHash":"0x3b8fb240d288781d4aac94d3fd16809ee413bc99294a085798a589dae51ddd4a",
+            "suggestedFeeRecipient":"0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b",
+            "stateRoot":"0xca3149fa9e37db08d1cd49c9061db1002ef1cd58db2210f2115c8c989b2bdf45",
+            "receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+            "logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
You could split this long line in more readable multiple lines using raw strings, e.g. see [test_types.cpp](https://github.com/torquem-ch/silkrpc/blob/master/silkrpc/json/types_test.cpp#L275)